### DISCUSSION TITLE
Como organizar minhas entidades e interfaces usando TS?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1789] [Perguntas e Respostas] [answered]
Fala galera, bom dia!

Estou iniciando meu primeiro projeto grande com React + TS (+Next +SASS).

Bom, tenho algumas interfaces como por exemplo a "IMovies" que uso em diversas partes do projeto, como vocês criam a estrutura de arquivos para separar estas entidades? Chegam a deixar entidades/interfaces direto no arquivo ou diretório dos componentes? Tenho algumas interfaces que uso apenas em certos componentes, como suas props...

E galera, alguém ai usando axios com TS, como faço para que o retorno já venha no formato de minha interface?

Muito obrigado!!

**Comment by waldemirflj** [comment, marked as answer]
Cara, não existe uma regra... Se fosse eu, eu criaria uma pasta interfaces na src e seguiria assim. E se fosse algo mais específico, deixaria junto ao contexto dessa necessidade.

> E galera, alguém ai usando axios com TS, como faço para que o retorno já venha no formato de minha interface?

Neste caso vc precisa um método/func envolvendo sua request devolvendo uma promise da sua interface.
